### PR TITLE
Add tests for `sf::MemoryInputStream`

### PR DESCRIPTION
**Pull request by Bambo-Borris**
## Description

`sf::FileInputStream` tests were already covered but noticed `sf::MemoryInputStream` tests were missing. Hoping these tests provide adequate coverage. 

## Tasks

* [ ] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?
CI runs the tests!

### CODE REVIEW

File: test/System/MemoryInputStream.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include <SFML/System/MemoryInputStream.hpp>
+#include "SystemUtil.hpp"
+
+#include <doctest.h>
+
+TEST_CASE("sf::MemoryInputStream class - [system]")
+{
+	SUBCASE("Empty stream")
```
Comment (kimci86) [inline test/System/MemoryInputStream.cpp:8]: `getSize()` should also be tested in this subcase.
Also, please indent with spaces instead of tabs.

Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp:8]: Yeah didn't notice it was set to tabs, I'll add a `getSize()` check and format it :)

File: test/System/MemoryInputStream.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#include <SFML/System/MemoryInputStream.hpp>
+#include "SystemUtil.hpp"
+
```
Comment (kimci86) [inline test/System/MemoryInputStream.cpp:3]: Maybe also include `<string_view>`

File: test/System/MemoryInputStream.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    SUBCASE("Temporary memory stream")
+    {
+        const std::string memoryContents = "hello world";
+        sf::MemoryInputStream mis;
+        mis.open(memoryContents.data(), sizeof(char) * memoryContents.size());
+
+        char buffer[32];
+        CHECK(mis.read(buffer, 5));
```
Comment (kimci86) [inline test/System/MemoryInputStream.cpp:25]:
```suggestion
        CHECK(mis.read(buffer, 5) == 5);
```

**Review by ChrisThrasher — APPROVED** [review]
Test coverage goes brrrrrr

File: test/System/MemoryInputStream.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        sf::MemoryInputStream mis;
+
+        CHECK(mis.read(nullptr, 0) == -1);
+        CHECK(mis.seek(0) == -1);
+        CHECK(mis.tell() == -1);
+        CHECK(mis.getSize() == -1);
+    }
+
+    SUBCASE("Temporary memory stream")
```
Comment (kimci86) [inline test/System/MemoryInputStream.cpp:18]: I am not sure "Temporary" is meaningful here.

Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp:18]: I took the case name from the `FileInputStream` one, could just change it to something like "Memory stream open" or something

File: test/System/MemoryInputStream.cpp (hunk 3 above)
Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp:25]: Should we not verify that the `mis.read()` actually returned the expected number of bytes read in this case?

File: test/System/MemoryInputStream.cpp (hunk 4 above)
Comment (kimci86) [reply test/System/MemoryInputStream.cpp:18]: In the context of `FileInputStream`, a temporary file is meaningful, but a temporary memory doesn't make so much sense. I think "Open memory stream" would be more meaningful here.

File: test/System/MemoryInputStream.cpp (hunk 3 above)
Comment (kimci86) [reply test/System/MemoryInputStream.cpp:25]: Yes, this is exactly what I suggested if I understand correctly.

Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp:25]: Oh wait yeah I misread, seems I'm reading my diffs wrong today. Yeah I'm perfectly happy with changing this.

File: test/System/MemoryInputStream.cpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#include <SFML/System/MemoryInputStream.hpp>
+#include <string_view>
+#include "SystemUtil.hpp"
```
Comment (kimci86) [inline test/System/MemoryInputStream.cpp:3]: One last thing, standard library headers come last by convention. See https://www.sfml-dev.org/style.php#includes
```suggestion
#include <SFML/System/MemoryInputStream.hpp>
#include "SystemUtil.hpp"
#include <string_view>
```

Comment (ChrisThrasher) [reply test/System/MemoryInputStream.cpp:3]: I don't even think `SystemUtil.hpp` is getting used.

Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp]: > I don't even think `SystemUtil.hpp` is getting used.

No seems not, I think it's safe to remove it from the includes

Comment (kimci86) [reply test/System/MemoryInputStream.cpp]: `<string>` is missing now :smile:
A raw array could also be used instead of the string.

Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp]: It appears it was how this file was getting `std::string` for the Linux GCC build so I'll also add the `<string>` include on

**Review by kimci86 — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: test/System/MemoryInputStream.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#include <SFML/System/MemoryInputStream.hpp>
+#include <string_view>
+#include <string>
+#include <sstream>
```
Comment (vittorioromeo) [inline test/System/MemoryInputStream.cpp:4]: I think we can avoid these two.

File: test/System/MemoryInputStream.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+        CHECK(mis.read(nullptr, 0) == -1);
+        CHECK(mis.seek(0) == -1);
+        CHECK(mis.tell() == -1);
+        CHECK(mis.getSize() == -1);
+    }
+
+    SUBCASE("Open memory stream")
+    {
+        const std::string memoryContents = "hello world";
```
Comment (vittorioromeo) [inline test/System/MemoryInputStream.cpp:22]:
```suggestion
        using std::literals::string_view_literals;
        constexpr auto memoryContents = "hello world"sv;
```

**Review by vittorioromeo — CHANGES_REQUESTED** [review]
Looks good, but I think we can avoid depending on `string` and `sstream`.

File: test/System/MemoryInputStream.cpp (hunk 7 above)
Comment (Bambo-Borris) [reply test/System/MemoryInputStream.cpp:22]: It's possible to make this change but you still require `<sstream>` to build for me locally so we'd only be dropping 1 include `<string>`

**Review by vittorioromeo — APPROVED** [review]

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

**Comment by vittorioromeo** [comment]
Thanks!

**Pull request merged**